dolphinscheduler-dao/src/main/resources/sql/upgrade/2.0.7_schema/postgresql/dolphinscheduler_ddl.sql

```diff
@@ -0,0 +1,20 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+drop index if exists task_instance_idx_code_pid_edtm;
+create index task_instance_idx_code_pid_edtm on t_ds_task_instance (task_code, process_instance_id, end_time);
```

Comment:
I think adding this has nothing to do with this issue. Please pull request separately

Reply:
it's a performance optimize if use this pr, on my platform, it have 5000+ dependent task to run, it generate 15000 records per day in t_ds_task_instance, in preview version, I added index (task_code, end_time)(2.0.6) and (process_instance_id, flag, start_time)(<=2.0.5), if don't create this index, a large number of full table scans will lead to huge resource consumption and the scheduler's database will be paralyzed.

Reply:
I agree with you, this index is important, but it has nothing to do with the current issue, please submit PR separately

Reply:
ok.

Reply:
which is new pr ?

Reply:
not create yet...
because... this index is only valid for this PR, if create alone, it has no effect on versions 2.0.0 to 2.0.6...

Reply:
OK, thanks

Reply:
I created a pr before, but not merged, no reason, it was auto closed:
#8290